File: src/solutions/easy/p217_contains_duplicate.rs
/*
Leetcode: https://leetcode.com/problems/contains-duplicate/
*/
#[cfg(test)]
fn contains_duplicate(nums: Vec<i32>) -> bool {
    use std::collections::HashSet;

    let mut num_set = HashSet::new();
    for num in nums {
        if num_set.contains(&num) {
            return true;
        }
        num_set.insert(num);
    }
    false
}

#[test]
fn test_contains_duplicate() {
    let test_cases: [(Vec<i32>, bool); 3] = [
        (vec![1, 2, 3, 1], true),
        (vec![1, 2, 3, 4], false),
        (vec![1, 1, 1, 3, 3, 4, 3, 2, 4, 2], true),
    ];

    for (case, want) in test_cases {
        assert_eq!(contains_duplicate(case), want);
    }
}

File: python/solutions/easy/p217_contains_duplicate.py
"""
Leetcode: https://leetcode.com/problems/contains-duplicate/
"""

from utils import TestCase


def contains_duplicate(nums):
    original_length = len(nums)
    length_of_set = len(set(nums))

    return original_length != length_of_set


def test_contains_duplicate():
    test_cases: list[TestCase] = [
        TestCase(input=[1, 2, 3, 1], expected=True),
        TestCase(input=[1, 2, 3, 4], expected=False),
        TestCase(input=[1, 1, 1, 3, 3, 4, 3, 2, 4, 2], expected=True),
    ]

    for case in test_cases:
        assert contains_duplicate(case.input) == case.expected
